packages/terra-popup/src/_PopupContent.jsx

```diff
@@ -0,0 +1,284 @@
+};
+
+/**
+ * Key code value for the escape key.
+ */
+const KEYCODES = {
+  ESCAPE: 27,
+};
+
+/**
+ * Valid positions for the arrow, applies a margin to the corresponding side.
+ */
+const ARROW_POSITIONS = [
+  'top',
+  'bottom',
+  'left',
+  'right',
+];
+
+/**
+ * Margin value used for calculations.
+ */
+const POPUP_MARGIN = 9;
+
+const propTypes = {
+  /**
+   * The arrow to be placed within the popup frame.
+   */
+  arrow: PropTypes.element,
+  /**
+   * The initial arrow position.
+   */
+  arrowPosition: PropTypes.oneOf(ARROW_POSITIONS),
+  /**
+   * The children to be presented as the popup's content.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * CSS classnames that are append to the popup content body.
+   */
+  classNameInner: PropTypes.string,
+  /**
+   * Whether or not the using the escape key should trigger the onRequestClose callback.
+   */
+  closeOnEsc: PropTypes.bool,
+  /**
+   * Whether or not clicking outside the popup should trigger the onRequestClose callback.
+   */
+  closeOnOutsideClick: PropTypes.bool,
+  /**
+   * Whether or not resizing the screen should trigger the onRequestClose callback.
+   */
+  closeOnResize: PropTypes.bool,
+  /**
+   * A the px value of height to be applied to the content container.
+   */
+  contentHeight: PropTypes.number.isRequired,
+  /**
+   * The maximum height to set for popup content in px, also used with responsive behavior for actual height.
+   */
+  contentHeightMax: PropTypes.number,
+  /**
+   * A the px value of width to be applied to the content container.
+   */
+  contentWidth: PropTypes.number.isRequired,
+  /**
+   * The maximum width of the popup content in px, also used with responsive behavior for actual width.
+   */
+  contentWidthMax: PropTypes.number,
+  /**
+   * Should the default behavior, that inserts a header when constraints are breached, be disabled.
+   */
+  isHeaderDisabled: PropTypes.bool,
+  /**
+   * The function that should be triggered when a close is indicated.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * The function returning the frame html reference.
+   */
+  refCallback: PropTypes.func,
+};
+
+const defaultProps = {
+  arrowPosition: 'top',
+  classNameInner: null,
+  closeOnEsc: false,
+  closeOnOutsideClick: false,
+  closeOnResize: false,
+  contentHeightMax: -1,
+  contentWidthMax: -1,
+  isHeaderDisabled: false,
+};
+
+class PopupContent extends React.Component {
+  static getContentStyle(height, maxHeight, width, maxWidth) {
+    const validHeight = maxHeight <= 0 || height <= maxHeight ? height : maxHeight;
+    const validWidth = maxWidth <= 0 || width <= maxWidth ? width : maxWidth;
+    return { height: `${validHeight.toString()}px`, width: `${validWidth.toString()}px` };
+  }
+
+  static addPopupHeader(children, onRequestClose) {
+    const icon = <IconClose tabIndex="0" className="terra-PopupContent-closeButton" onClick={onRequestClose} height="30" width="30" style={{ float: 'right' }} />;
```

Comment:
If you use a button with the variant set to `link`, I believe it should look the same as you have it now. Might need some CSS to adjust the color.